source/overloading.tex

```diff
@@ -1529,21 +1530,84 @@
 \end{itemize}
 \end{itemize}
 
+\pnum
+When resolving a placeholder for a deduced class type\iref{dcl.type.simple}
+where the \grammarterm{template-name} names an alias template \tcode{A}
+whose \grammarterm{defining-type-id}
+is a \grammarterm{simple-template-id} \tcode{B<L>},
+the guides of \tcode{A} are the set of function or function templates
+formed as follows.
+For each function or function template \tcode{f} in the guides of \tcode{B},
+form a function or function template \tcode{f'}
+according to the following procedure and add it to the set:
+\begin{itemize}
+\item
+  Deduce the template arguments of the return type of \tcode{f}
+  from \tcode{B<L>} according to the process in\iref{temp.deduct.type}
+  with the exception that deduction does not fail
+  if not all template arguments are deduced.
+  Let \tcode{g} denote the result of substituting
+  these deductions into \tcode{f}.
+  If the substitution fails, no \tcode{f'} is produced.
+  Form the function or function template \tcode{f'} as follows:
+  \begin{itemize}
+  \item
+    The function type of \tcode{f'} is the function type of \tcode{g}.
+  \item
+    If \tcode{f} is a function template,
+    the template parameter list of \tcode{f'} consists of
+    all the template parameters of \tcode{A}
+    (including their default template arguments)
+    that appear in the above deductions or
+    (recursively) in their default template arguments,
+    followed by the template parameters of \tcode{f} that were not deduced
+    (including their default template arguments).
+  \item
+    The associated constraints\iref{temp.constr.decl} are
+    the conjunction of the associated constraints of \tcode{g} and
+    a constraint that is satisfied if and only if
+    the arguments of \tcode{A} are deducible (see below) from the return type.
+  \end{itemize}
+\item
+  If \tcode{f} is a copy deduction candidate\iref{over.match.class.deduct},
+  then \tcode{f'} is considered to be so as well.
+\item
+  If \tcode{f} was generated
+  from a deduction-guide\iref{over.match.class.deduct},
+  then \tcode{f'} is considered to be so as well.
+\item
+  The \grammarterm{explicit-specifier} of \tcode{f'} is
+  the \grammarterm{explicit-specifier} of g (if any).
+\end{itemize}
+
+\pnum
+The arguments of a template \tcode{A} are said to be
+deducible from a type \tcode{T} if, given a class template
+\begin{codeblock}
+template <typename> class AA;
+\end{codeblock}
+with a single partial specialization
+whose template parameter list is that of \tcode{A} and
+whose template argument list is a specialization of \tcode{A}
+with the template argument list of \tcode{A}\iref{temp.dep.type},
+\tcode{AA<T>} matches the partial specialization.
```

Comment:
Look at the example below. We're looking at a partial specialization like `AA<A<something>>`, and deduce `AA<T>` against it. Yes, it would be nicer to just branch off to some general deduction facility and discuss P and A here, but we don't.

Reply:
Can we move the example up here?  It nice to have examples to help clarify the normative wording.

We do have this big monolithic example at the end - I'd like to see it split up and moved closer to the normative wording it applies to.

I know @zygoloid needs to have this merged asap - if there's not time for this please open an issue for it, as well as for my issue with "match".

Reply:
I'd suggest you e-mail Mike Miller to open a core issue for "match", and also ask for an example in the proximity. The example at the bottom is a hypothetical rewrite that we can't easily split.

Reply:
issue reported to Mike